designs/odcr.md

```diff
@@ -0,0 +1,213 @@
+- Supporting conditions for capacity reservation instance creations. _By showing the available instance count within each capacity reservation for an EC2NodeClass we have similar information available_
+
+## Proposed Solution
+
+### Supporting associating Capacity Reservations to EC2NodeClass
+
+- Add a new field under `spec` for `capacityReservationSelectorTerms` to `EC2NodeClass` for defining which Capacity Reservation to be used for a specific `EC2NodeClass`
+  - This will allow us to attach multiple Capacity Reservations across AZs and Instance Types to a single EC2NodeClass. This capability removes the need for Capacity Reservation Groups for this MVP.
+- Add a new field under `status` for the found Capacity Reservations by the `spec.capacityReservationSelectorTerms` for the `EC2NodeClass`
+
+```yaml
+apiVersion: karpenter.k8s.aws/v1beta1
+kind: EC2NodeClass
+metadata:
+  name: example-node-class
+spec:
+  capacityReservationSelectorTerms:
+    - # The Availability Zone of the Capacity Reservation
+      availabilityZone: String | None
+      # The platform of operating system for which the Capacity Reservation reserves capacity
+      id: String | None
+      # The type of operating system for which the Capacity Reservation reserves capacity
+      instanceType: String | None
+      # The ID of the Amazon Web Services account that owns the Capacity Reservation
+      ownerId: String | None
+      # Tags is a map of key/value tags used to select subnets
+      # Specifying '*' for a value selects all values for a given tag key.
+      tags: Map | None
+      # Indicates the type of instance launches that the Capacity Reservation accepts. The options include:
+      #    - open:
+      #       The Capacity Reservation accepts all instances that have
+      #       matching attributes (instance type, platform, and Availability
+      #       Zone). Instances that have matching attributes launch into the
+      #       Capacity Reservation automatically without specifying any
+      #       additional parameters.
+      #    - targeted:
+      #       The Capacity Reservation only accepts instances that
+      #       have matching attributes (instance type, platform, and
+      #       Availability Zone), and explicitly target the Capacity
+      #       Reservation. This ensures that only permitted instances can use
+      #             the reserved capacity.
+      type: String | None
+status:
+  capacityReservations:
+    - # AvailabilityZone of the Capacity Reservation
+      availabilityZone: String
+      # Available Instance Count of the Capacity Reservation
+      availableInstanceCount: Integer
+      # The date and time at which the Capacity Reservation expires. When a Capacity
+      # Reservation expires, the reserved capacity is released and you can no longer
+      # launch instances into it. The Capacity Reservation's state changes to expired
+      # when it reaches its end date and time.
+      endDate: String | None
```

Comment:
```suggestion
      endTime: String | None
```

Kubernetes standard is to use `time` when denoting time-based fields. See things like `lastSuccessfulTime` in CronJob or `lastTransitionTime` in all status conditions